src/sql/src/statements/create.rs

```diff
@@ -229,6 +265,68 @@ pub struct CreateExternalTable {
     pub engine: String,
 }
 
+impl CreateExternalTable {
+    fn format_constraints(&self) -> String {
+        self.constraints
+            .iter()
+            .map(|c| {
+                if is_time_index(c) {
+                    let TableConstraint::Unique { columns, .. } = c else {
+                        unreachable!()
+                    };
+
+                    format_indent!("{}TIME INDEX ({})", format_list_comma!(columns))
+                } else {
+                    format_indent!(c)
+                }
+            })
+            .join(LINE_SEP)
+    }
+
+    #[inline]
+    fn format_if_not_exists(&self) -> &str {
+        if self.if_not_exists {
+            "IF NOT EXISTS"
+        } else {
+            ""
+        }
+    }
+
+    #[inline]
+    fn format_options(&self) -> String {
```

Comment:
ditto for all these options. We may make a util for formatting or redacting options (map sensitive_key=value to sensitive_key="***") however.